src/set_theory/surreal.lean

```diff
@@ -1,199 +1,29 @@
 /-
 Copyright (c) 2019 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Mario Carneiro
-
-The basic theory of surreal numbers.
+Authors: Mario Carneiro, Scott Morrison
 -/
-import tactic.interactive data.nat.cast
-
-/-- The type of pre-surreal numbers, before we have quotiented
-  by extensionality. In ZFC, a surreal number is constructed from
-  two sets of surreal numbers that have been constructed at an earlier
-  stage. To do this in type theory, we say that a pre-surreal is built
-  inductively from two families of pre-surreals indexed over any type
-  in Type u. The resulting type `pSurreal.{u}` lives in `Type (u+1)`,
-  reflecting that it is a proper class in ZFC. -/
-inductive {u} pSurreal : Type (u+1)
-| mk : ∀ α β : Type u, (α → pSurreal) → (β → pSurreal) → pSurreal
-
-namespace pSurreal
-
-/-- Define simultaneously by mutual induction the `<=` and `<`
-  relation on surreals. The ZFC definition says that `x = {xL | xR}`
-  is less or equal to `y = {yL | yR}` if `∀ x₁ ∈ xL, x₁ < y`
-  and `∀ y₂ ∈ yR, x < y₂`, where `x < y` is the same as `¬ y <= x`.
-  This is a tricky induction because it only decreases one side at
-  a time, and it also swaps the arguments in the definition of `<`.
-  So we define `x < y` and `x <= y` simultaneously. -/
-def le_lt (x y : pSurreal) : Prop × Prop :=
-begin
-  induction x with xl xr xL xR IHxl IHxr generalizing y,
-  induction y with yl yr yL yR IHyl IHyr,
-  exact ((∀ i, (IHxl i ⟨yl, yr, yL, yR⟩).2) ∧ (∀ i, (IHyr i).2),
-         (∃ i, (IHxr i ⟨yl, yr, yL, yR⟩).1) ∨ (∃ i, (IHyl i).1))
-end
-
-instance : has_le pSurreal := ⟨λ x y, (le_lt x y).1⟩
-instance : has_lt pSurreal := ⟨λ x y, (le_lt x y).2⟩
+import set_theory.game
 
-/-- Definition of `x ≤ y` on pre-surreals. -/
-@[simp] theorem mk_le_mk {xl xr xL xR yl yr yL yR} :
-  (⟨xl, xr, xL, xR⟩ : pSurreal) ≤ ⟨yl, yr, yL, yR⟩ ↔
-  (∀ i, xL i < ⟨yl, yr, yL, yR⟩) ∧
-  (∀ i, (⟨xl, xr, xL, xR⟩ : pSurreal) < yR i) := iff.rfl
+/-!
```

Comment:
Could you give this file a full module doc?

Reply:
Done.